lib/cli/src/commands/app/list.rs

```diff
@@ -32,15 +41,37 @@ impl AsyncCliCommand for CmdAppList {
     async fn run_async(self) -> Result<(), anyhow::Error> {
         let client = self.api.client()?;
 
-        let apps = if let Some(ns) = self.namespace {
-            wasmer_api::query::namespace_apps(&client, &ns).await?
+        let apps_stream: Pin<
+            Box<dyn Stream<Item = Result<Vec<DeployApp>, anyhow::Error>> + Send + Sync>,
+        > = if let Some(ns) = self.namespace.clone() {
+            Box::pin(wasmer_api::query::namespace_apps(&client, ns).await)
         } else if self.all {
-            wasmer_api::query::user_accessible_apps(&client).await?
+            Box::pin(wasmer_api::query::user_accessible_apps(&client).await?)
         } else {
-            wasmer_api::query::user_apps(&client).await?
+            Box::pin(wasmer_api::query::user_apps(&client).await)
         };
 
-        println!("{}", self.fmt.format.render(&apps));
+        let mut apps_stream = std::pin::pin!(apps_stream);
+
+        let mut rem = self.max;
+
+        let mut display_apps = vec![];
+
+        while let Some(apps) = apps_stream.next().await {
+            let mut apps = apps?;
+
+            let limit = std::cmp::min(apps.len(), rem);
+
+            if limit == 0 {
+                break;
+            }
+
+            display_apps.extend(apps.drain(..limit));
+
+            rem -= limit;
+        }
```

Comment:
I'd do this conditionally, kind of asking the user if they want to go to the next page.

Reply:
I think it's going to be annoying if the user wants to list 30 apps and has to type `y` every 10 apps. Maybe a dedicated flag like `--paging-mode` which asks at the end of each page?